### ISSUE TITLE
Backward pass returns Nan after the first batch

### ISSUE THREAD

**Issue by arnaghosh** [msamribeiro/deep-cca#1]
Hi,
I was working on a PyTorch version of your theano code of the cca loss. 
I was using a deep network to get features from X1, which gives me the data1 and the data2 is an unperturbed version of X2. The aim is to maximize the correlation between data1 and data2, so basically learn a transformation of X1 (using the deep network) that maximizes its correlation with X2. 
However, the backward pass returned Nan in the 'SymeigBackward' function (this is in PyTorch, the backprop function for symeig function. It is supposed to work the same as the T.nlinalg.eigh function of Theano). This problem gets somewhat resolved if I decrease the regularization parameter cca_reg1. But I am not sure why this happens. Any help would be much appreciated.
Also, even if the backward doesn't throw Nans the corr loss doesn't seem to reduce. I guess this might be something else though. (Just mentioning in case it might help understand what's going on). 
Thanks a bunch :)

**Comment by msamribeiro** [comment]
I apologise, but for some reason I missed this when it was first opened and I only checked it now. I hope you managed to solve the issue.
